gradle-conjure/src/main/java/com/palantir/gradle/conjure/GenerateNpmRcTask.java

```diff
@@ -0,0 +1,148 @@
+    private final RegularFileProperty outputFile = getProject().getObjects().fileProperty();
+    private final Property<String> username = getProject().getObjects().property(String.class);
+    private final Property<String> password = getProject().getObjects().property(String.class);
+    private final Property<String> packageName = getProject().getObjects().property(String.class);
+
+    @OutputFile
+    public final RegularFileProperty getOutputFile() {
+        return this.outputFile;
+    }
+
+    @Input
+    public final Property<String> getUsername() {
+        return username;
+    }
+
+    @Input
+    public final Property<String> getPassword() {
+        return password;
+    }
+
+    @Input
+    public final Property<String> getPackageName() {
+        return packageName;
+    }
+
+    // mainly for tests and should not be included in the generator options
+    private String getNpmRegistryUri() {
+        String userRegistry = getProject().hasProperty(NPM_REGISTRY_URI_PROPERTY)
+                ? getProject().property(NPM_REGISTRY_URI_PROPERTY).toString()
+                : "https://registry.npmjs.org";
+        return userRegistry.endsWith("/") ? userRegistry.substring(0, userRegistry.length() - 1) : userRegistry;
+    }
+
+    @TaskAction
+    public final void createNpmrc() throws IOException, InterruptedException {
+        int slashIndex = packageName.get().indexOf("/");
+        Optional<String> scope = packageName.get().startsWith("@") && slashIndex != -1
+                ? Optional.of(packageName.get().substring(1, slashIndex))
+                : Optional.empty();
+
+        String registryUri = getNpmRegistryUri();
+        String strippedUri = registryUri.startsWith("https://") ? registryUri.substring(8) : registryUri.substring(7);
+        NpmTokenResponse npmTokenResponse = tokenFromCreds(registryUri, username.get(), password.get());
+
+        String scopeRegistry = scope.map(s -> s + ":").orElse("");
+        String npmRcContents = String.format(
+                "%sregistry=%s\n//%s/:_authToken=%s",
+                scopeRegistry, registryUri, strippedUri, npmTokenResponse.token());
+
+        try {
+            Files.writeString(outputFile.getAsFile().get().toPath(), npmRcContents, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static NpmTokenResponse tokenFromCreds(String registryUri, String username, String password)
+            throws IOException, InterruptedException {
+        HttpResponse<NpmTokenResponse> response = HttpClient.newHttpClient()
+                .send(
+                        HttpRequest.newBuilder()
+                                .header(HttpHeaders.ACCEPT, "application/json")
+                                .header(HttpHeaders.CONTENT_TYPE, "application/json")
+                                .uri(URI.create(String.format("%s/-/user/org.couchdb.user:%s", registryUri, username)))
```

Comment:
That's the npm api :/

Reply:
https://stackoverflow.com/questions/23460980/how-to-set-npm-credentials-using-npm-login-without-reading-from-stdin was useful deciphering what to do here. Otherwise you want to look into npm-registry-client and look at `adduser` method

Reply:
:homerhedge: ahh, leaky abstractions... continue as you were